src/unix/core.c

```diff
@@ -983,3 +984,61 @@ int uv__dup2_cloexec(int oldfd, int newfd) {
     return r;
   }
 }
+
+
+int uv_os_homedir(char* buffer, size_t* size) {
+  struct passwd pw;
+  struct passwd* result;
+  char* buf;
+  uid_t uid;
+  size_t bufsize;
+  size_t len;
+  int r;
+
+  if (buffer == NULL || size == NULL || *size == 0)
+    return -EINVAL;
+
+  bufsize = sysconf(_SC_GETPW_R_SIZE_MAX);
+
+  if (bufsize <= 0)
+    return -EIO;
+
+  uid = getuid();
```

Comment:
Alright, on your head be it.  We can enforce serial access in io.js so it's not a big problem for us but the documentation should come with a big fat warning that `uv_os_homedir()` is not MT-safe because of the `getenv()` call.

@cjihrig Would you mind making the requested changes?